modules/app-connector/src/server/kyma/event.ts

```diff
@@ -26,8 +29,68 @@ export function send(event: any): Promise<any> {
                 LOGGER.debug("Received send confirmation: %s", JSON.stringify(response.body, ["id", "name"], 2))
                 return response.body;
             } else {
-                throw common.resolveError(response.statusCode, response.body, "sending event")
+                throw common.resolveError(response.statusCode, response.body, "sending legacy event")
             }
         })
     })
 }
+
+export function sendCloudEvent(event: any): Promise<any> {
+    return connection.cloudEventsUrl().then((cloudEventsUrl) => {
+        LOGGER.debug("Received Body: " + JSON.stringify(event, null, 2));
+        let headers: any = {
+            "Content-Type": "application/cloudevents+json"
+        }
+        if (event["eventtracing"]) {
+            headers["x-b3-sampled"] = "1"
+        }
+        LOGGER.debug("Sending Header: " + JSON.stringify(headers, null, 2));
+        return request(common.createRequestOptions(connection, {
+            uri: cloudEventsUrl, 
+            method: "POST",
+            headers: headers,
+            body: event
+        })).then((response: any) => {
+            if (response.statusCode < 300) {
+                LOGGER.debug("Received send confirmation: %s", JSON.stringify(response.body, null, 2))
+                return response.body;
+            } else {
+                throw common.resolveError(response.statusCode, response.body, "sending cloud event in structured mode")
+            }
+        })
+    })
+}
+
+export function sendCloudEventBinary(body: any, header: any): Promise<any> {
+    return connection.cloudEventsUrl().then((cloudEventsUrl) => {
+        LOGGER.debug("Received Body: " + JSON.stringify(body, null, 2) + " and Header: " + JSON.stringify(header, null, 2));
+        let spec = header["ce-specversion"];

```

Comment:
Could be maybe inlined instead of having dedicated variable assignment, don't see an improvement in readability. But that's just a matter of taste, feel free to ignore :)

Reply:
I think so too, inlined it.